### ISSUE TITLE
Bump minimum OS versions

### ISSUE THREAD

**Issue by philipphofmann** [getsentry/sentry-cocoa#3846]
### Description

As you soon can only publish to the App Store with Xcode 15, we can bump the minimum OS of the SDK to the versions [supported by Xcode 15](https://developer.apple.com/documentation/xcode-release-notes/xcode-15-release-notes), which are iOS 12 and later, tvOS 12 and later, and watchOS 4 and later. 

We don't need a major bump for this, but we should clearly point this out in the Changelog and the docs.

**Comment by armcknight** [comment]
Continuing the discussion here on this comment: https://github.com/getsentry/sentry-cocoa/pull/3845#pullrequestreview-1993989747

> Xcode 15 allows you to target macOS 10.13, iOS 12, tvOS 12, and watchOS 4. We must support these versions. This is also in our [SDK Philosophy and Design Principles](https://develop.sentry.dev/sdk/philosophy/#compatibility-is-king).

I am on board with compatibility, be we have to be able to test the platforms we claim to support. We should either prioritize getting that working, or compromise on the stance. it can still be a guideline without being absolute dogma. I think if we must choose between compromising on validated releases via automated testing, or compromising on supporting the earliest possible platform version, then automated testing should win out over compatibility.

I think this philosophy is covered in the section right above it, under "Degrade Gracefully":
> If this cannot be achieved we need to make sure that we inform the user ahead of time by alerting about such situations at installation time. If for instance an SDK requires a minimum version to even build, we need to ensure that the user is informed and that we do not cause confusing errors later.

IMO This is why this should be a breaking change with a bump to 9.0.0, and then in the highly unlikely event we hear from a customer that they need either a new feature or a patch on e.g. iOS <13, we could backport a hotfix to 8.x.y. That, to me, is how you really hit the sweet spot between Compatibility and Graceful Degradation.

**Comment by bruno-garcia** [comment]
I love these principles but Compatibility is King doesn't mean support stuff no one uses.

**Comment by philipphofmann** [comment]
> I am on board with compatibility, be we have to be able to test the platforms we claim to support.

It's essential we test our code on the most used / most recent platforms, of course, but I think it's acceptable not to test on all platforms we support, especially the ones almost nobody uses anymore. The major reason why we don't test our code on iOS 12 and such is because GH actions doesn't have good support for that. With our test suite in place, it mayn't reveal bugs on older, less-used OS versions, but that chance also exists on the most adopted OS versions.

> So I'd consider the data such as "how many orgs are sending data from iOS versions we're aiming to drop, on newer versions of the SDK" into account as opposed to using that as a blank card to never drop outdated versions

Even if we don't receive much data for outdated iOS versions, dropping OS version support could be a dealbreaker for some customers. For example, [TikTok](https://apps.apple.com/us/app/tiktok/id835599320) and [What's App](https://apps.apple.com/us/app/whatsapp-messenger/id310633997) still target iOS 12. While those two might never consider Sentry because they tend to build everything themselves, it would be bad for us if a similar customer still targeting iOS 12 doesn't want to use us because our latest SDK version no longer supports them.

What do we gain if we drop iOS 11 and 12 support and bump to iOS 13?

**Comment by kahest** [comment]
My stab at a pragmatic "ruleset"

- On a major release of the SDK, we bump the minimum versions to the supported versions of the oldest Xcode that can be used to submit to App Store
- When there's a discrepancy between that and what we can/do test on CI, we point that out in release notes. E.g. we **support** iOS 13+ because we test it on CI, and we **allow on a best-effort basis** 12 because it's still expected to work and we will provide fixes if necessary (that we can test & verify outside of CI)

We should be able to stick to that unless there's a specific reason not to. E.g. an older OS version would require significant effort to keep supporting, or there are significant workarounds in place for a OS version that are not warranted to keep around because of low adoption of the OS. Then we just discontinue support for this version altogether.

**Comment by armcknight** [comment]
Yeah, I like the idea of "best-effort" vs "full" support, like some of the other platforms we looked at. I'm cool with doing things that way. Then our test targets could have a higher deployment target to be able to take advantage of newer API and dependencies, while our SDK is still available to those that need earlier platform version support.

I'm find closing my PR and waiting for the next major. Generally, we should plan to at least release a new major with min support bumps on each new Apple OS release every fall.

**Comment by kahest** [comment]
Note on 

> On a major release of the SDK, we bump the minimum versions to the **supported** versions of the oldest Xcode that can be used to submit to App Store

Currently, Xcode 15 + iOS17 is required to submit to app store. Xcode 15 **supports iOS12 and newer**, while iOS18 is current major. I don’t think we need to support iOS12 (pending internal data confirmation), iOS14 sounds valid, as discussed internally. Maybe major-4 is the rule of thumb for the time being.

**Comment by armcknight** [comment]
I recommend we close this issue until we're ready to do the next bump. See also #4971 to capture a decision on this that can govern subsequent periodic bumps.

**Comment by kahest** [comment]
We can just keep this issue open IMO, it's already in the v9 milestone and we should start working towards v9 now-ish
